File: app/src/core/utility/base64Converter.test.ts
import { describe, expect, it } from 'vitest'
import { Base64Converter } from './base64Converter'

describe('Base64Converter', () => {
  const converter = new Base64Converter()

  it('encodes text to Base64', () => {
    const result = converter.execute({ text: 'InfraKit Studio', operation: 'encode' })
    expect(result.output).toBe('SW5mcmFLaXQgU3R1ZGlv')
  })

  it('decodes Base64 back to text', () => {
    const result = converter.execute({ text: 'SW5mcmFLaXQgU3R1ZGlv', operation: 'decode' })
    expect(result.output).toBe('InfraKit Studio')
  })

  it('decodes Base64 pasted without padding', () => {
    const result = converter.execute({ text: 'SGVsbG8', operation: 'decode' })
    expect(result.output).toBe('Hello')
  })

  it('rejects invalid Base64 input', () => {
    expect(() =>
      converter.execute({ text: '@@@not base64@@@', operation: 'decode' }),
    ).toThrow()
  })

  it('rejects empty decode input', () => {
    expect(() => converter.execute({ text: '   ', operation: 'decode' })).toThrow()
  })
})

File: archive/flutter-app/test/core/utility/base64_converter_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:infrakit_studio/core/utility/base64_converter.dart';

void main() {
  const converter = Base64Converter();

  test('encodes text to Base64', () {
    final result = converter.execute(
      const Base64ConversionInput(text: 'InfraKit Studio', operation: Base64Operation.encode),
    );
    expect(result.output, 'SW5mcmFLaXQgU3R1ZGlv');
  });

  test('decodes Base64 back to text', () {
    final result = converter.execute(
      const Base64ConversionInput(text: 'SW5mcmFLaXQgU3R1ZGlv', operation: Base64Operation.decode),
    );
    expect(result.output, 'InfraKit Studio');
  });

  test('decodes Base64 pasted without padding', () {
    final result = converter.execute(
      const Base64ConversionInput(text: 'SGVsbG8', operation: Base64Operation.decode),
    );
    expect(result.output, 'Hello');
  });

  test('rejects invalid Base64 input', () {
    expect(
      () => converter.execute(
        const Base64ConversionInput(text: '@@@not base64@@@', operation: Base64Operation.decode),
      ),
      throwsFormatException,
    );
  });

  test('rejects empty decode input', () {
    expect(
      () => converter.execute(const Base64ConversionInput(text: '   ', operation: Base64Operation.decode)),
      throwsArgumentError,
    );
  });
}
